packages/SOLID/src/Rector/ClassConst/PrivatizeLocalClassConstantRector.php

```diff
@@ -92,17 +101,50 @@ public function refactor(Node $node): ?Node
 
         /** @var string $constant */
         $constant = $this->getName($node);
+        $classNode = $this->parsedNodesByType->findClass($class);
+        $mustBeAtLeastProtected = false;
+
+        if ($classNode !== null && $classNode->hasAttribute(AttributeKey::PARENT_CLASS_NAME)) {
+            /** @var string $parentClassName */
+            $parentClassName = $classNode->getAttribute(AttributeKey::PARENT_CLASS_NAME);
+            if ($parentClassName) {
+                $parentClassConstant = $this->parsedNodesByType->findClassConstant($parentClassName, $constant);
+                if ($parentClassConstant) {
+                    $this->refactor($parentClassConstant);
+
+                    // The parent's constant is public, so this one must become public too
+                    if ($parentClassConstant->isPublic()) {
+                        $this->makePublic($node);
+                        return $node;
+                    }
+
+                    // The parent's constant is protected, so this one must become protected or weaker
+                    if ($parentClassConstant->isProtected()) {
+                        $mustBeAtLeastProtected = true;
+                    }
+                }
+            }
+        }
```

Comment:
This should be decouple to own method, so code is better readable e.g. `resolveMinimalParentAllowedVisbilityLevel()`, returning `public`, `protected` or `private` string

Reply:
Refactored for better readability.

Reply:
Thanks